File: src/pattern/mediator/java/ColleagueButton.java
package pattern.mediator.java;

import java.awt.*;

public class ColleagueButton extends Button implements Colleague{
    private Mediator mediator;

    public ColleagueButton(String title){
        super(title);
    }
    @Override
    public void setMediator(Mediator mediator) {
        this.mediator = mediator;
    }

    @Override
    public void setColleagueEnabled(boolean enabled) {
        super.setEnabled(enabled);
    }
}

File: src/pattern/mediator/kotlin/ColleagueButton.kt
package pattern.mediator.kotlin

import java.awt.Button

class ColleagueButton(title: String): Button(title), Colleague{
    lateinit var _mediator: Mediator



    override fun setMediator(mediator: Mediator) {
        this._mediator = mediator
    }

    override fun setColleagueEnabled(enabled: Boolean) {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }


}